controllers/goalertintegration/clusterdeployment_created.go

```diff
@@ -1,11 +1,191 @@
+	)
+
+	if !util.ContainsFinalizer(cd, finalizer) {
+		baseToPatch := client.MergeFrom(cd.DeepCopy())
+		util.AddFinalizer(cd, finalizer)
+		return r.Patch(context.TODO(), cd, baseToPatch)
+	}
+
+	clusterID := GetClusterID(cd)
+
+	// Load data to create new service in Goalert
+	dataHighSvc := &goalert.Data{
+		EscalationPolicyID: gi.Spec.HighEscalationPolicy,
+		Name:               clusterID + " - High",
+		Description:        cd.Spec.ClusterName,
+		Favorite:           true,
+	}
+
+	dataLowSvc := &goalert.Data{
+		EscalationPolicyID: gi.Spec.LowEscalationPolicy,
+		Name:               clusterID + " - Low",
+		Description:        cd.Spec.ClusterName,
+		Favorite:           true,
+	}
+
+	highSvcID, err := gclient.CreateService(dataHighSvc, sessionCookie)
+	if err != nil {
+		r.reqLogger.Error(err, "Failed to create service for High alerts")
+		return err
+	}
+	lowSvcID, err := gclient.CreateService(dataLowSvc, sessionCookie)
+	if err != nil {
+		r.reqLogger.Error(err, "Failed to create service for Low alerts")
+		return err
+	}
+
+	// Load data to create integration key for alertmanager
+	dataIntKeyHighSvc := &goalert.Data{
+		Id:   highSvcID,
+		Type: "prometheusAlertmanager",
+		Name: "High alerts",
+	}
+	dataIntKeyLowSvc := &goalert.Data{
+		Id:   lowSvcID,
+		Type: "prometheusAlertmanager",
+		Name: "Low alerts",
+	}
+
+	highIntKeyID, err := gclient.CreateIntegrationKey(dataIntKeyHighSvc, sessionCookie)
+	if err != nil {
+		r.reqLogger.Error(err, "Failed to create integration key for high alerts")
+		return err
+	}
+	lowIntKeyID, err := gclient.CreateIntegrationKey(dataIntKeyLowSvc, sessionCookie)
+	r.reqLogger.Error(err, "Failed to create integration key for low alerts")
+	if err != nil {
+		return err
+	}
+
+	// Load data to create heartbeatmonitor
+	dataHeartbeatMonitor := &goalert.Data{
+		Id:      highSvcID,
+		Name:    clusterID,
+		Timeout: 15,
+	}
+
+	heartbeatMonitorID, err := gclient.CreateHeartbeatMonitor(dataHeartbeatMonitor, sessionCookie)
+	if err != nil {
+		r.reqLogger.Error(err, "Failed to create heartbeat monitor")
+		return err
+	}
+
+	// save config map
+	newCM := kube.GenerateConfigMap(cd.Namespace, configMapName, highSvcID, lowSvcID, highIntKeyID, lowIntKeyID, highEscalationPolicyID, lowEscalationPolicyID)
+	if err = controllerutil.SetControllerReference(cd, newCM, r.Scheme); err != nil {
+		r.reqLogger.Error(err, "Error setting controller reference on configmap")
+		return err
+	}
+
+	if err := r.Create(context.TODO(), newCM); err != nil {
+		if errors.IsAlreadyExists(err) {
+			if updateErr := r.Update(context.TODO(), newCM); updateErr != nil {
+				r.reqLogger.Error(err, "Error updating existing configmap", "Name", configMapName)
+				return err
+			}
+			return nil
+		}
+		r.reqLogger.Error(err, "Error creating configmap", "Name", configMapName)
+		return err
+	}
+
+	//add secret part
+	secret := kube.GenerateGoalertSecret(cd.Namespace, secretName, highIntKeyID, lowIntKeyID, heartbeatMonitorID)
```

Comment:
Hey Katherine, It looks like we are adding `highIntKeyID` and `lowIntKeyID` to a secret as well as a configmap  earlier on (line 113). This logic makes sense in PD where there is a Key ID and the actual key value, but looking at GoAlert, it looks like its just a key value which is a URL including a token. Sounds like we don't want those values in a configmap for security reasons. Could we avoid putting that data into the configmap? 

The requirement for the configmap and the Key ID may only be a PagerDuty thing and might be unneeded for GoAlert. Looking at your [goalert service file](https://github.com/katherinelc321/configure-goalert-operator/blob/handle-create/pkg/goalert/service.go) we may only need the service id's and escalation policy id's in that configmap. Mind taking a look at this? Thanks and great job!